src/main/ruby/truffleruby/core/truffle/polyglot.rb

```diff
@@ -34,6 +34,157 @@ def self.as_enumerable(object)
     Truffle::Interop.enumerable(object)
   end
 
+  # HasArrayElementsTrait
+  module HasArrayElementsTrait
+
+    private def read_array_element(index)
+      Truffle::Interop.read_array_element(self, index)
+    end
+
+    def [](index)
+      index += length if index < 0
+      return nil if (index < 0 || index >= length)
+      read_array_element(index)
+    end
+
+    alias_method :at, :[]
+
+    def first
+      read_array_element(0)
+    end
+
+    def last
+      read_array_element(length - 1)
+    end
+
+    def size
+      Truffle::Interop.array_size(self)
+    end
+
+    alias_method :length, :size
+
+    def count(item = nil)
+      seq = 0
+      if (!item.nil?)
+        each { |o| seq += 1 if item == o }
+      elsif block_given?
+        each { |o| seq += 1 if yield(o) }
+      else
+        return size
+      end
+      seq
+    end
+
+    def to_a
+      if Truffle::Interop.has_array_elements?(self)
+        Truffle::Interop.to_array(self)
+      else
+        raise RuntimeError
+      end
+    end
+
+    alias_method :to_ary, :to_a
+
+    def each(*args)
+      return enum_for(:each) unless block_given?
+      i = 0
+      while i < length
+        yield read_array_element(i)
+        i += 1
+      end
+    end
+
+    def each_with_index
+      return enum_for(:each) unless block_given?
+      i = 0
+      while i < length
+        yield read_array_element(i), i
+        i += 1
+      end
+    end
+
+  end
+
+  # ForeignObject class
+  class ForeignObject < Object
+
+    def respond_to?(symbol, include_all = false)
+      Truffle::Interop.foreign_respond_to?(self, symbol)
+    end
+
+    def class
+      Truffle::Interop.foreign_class(self)
+    end
+
+    def object_id
+      Truffle::Interop.identity_hash_code(self)
+    end
+
+    alias_method :__id__, :object_id
```

Comment:
Style: `alias_method` should be directly below the method it aliases (no blank line in between), to make it even easier to spot which method it aliases.